app/models/protocol.rb

```diff
@@ -129,6 +129,10 @@ def is_study?
   def virgin_project?
     selected_for_epic == nil
   end
+  # returns true if selected_for_epic is nil in the database
+  def selected_for_epic_is_nil_in_db?
```

Comment:
In sub_service_requests_controll#update_from_project_study_information, if the set of attributes passed in from the client is invalid for some reason and @protocol.update_attributes fails, we need to maintain the state of the @protocol record so the values the user input are maintained when the page is re-rendered. The value of @protocol.selected_for_epic is being set to true/false in memory though, rather than staying as nil, so the view is rendered incorrectly (displaying the epic box as readonly, rather than making it editable). My solution was therefore to just load the value of @protocol._selected_for_epic from the database instead of relying on the possibly (incorrectly) true/false value in memory. I guess this isn't ideal and I think I found a way around it using some built in ActiveRecord methods.